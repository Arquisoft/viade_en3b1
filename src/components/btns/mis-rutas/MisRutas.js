import React from 'react';
import Button from '@material-ui/core/Button';
import { MenuItem, Menu, Paper, Popper, Grow, MenuList, Typography, Link } from '@material-ui/core';
import ListIcon from '@material-ui/icons/List';
import EditLocationIcon from '@material-ui/icons/EditLocation';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import { withStyles } from '@material-ui/core/styles';

const MisRutas = () => {

    const [anchorEl, setAnchorEl] = React.useState(null);

    const handleClick = (event) => {
        setAnchorEl(event.currentTarget);
    };

    const handleClose = () => {
        setAnchorEl(null);
    };

    return (
        <div>
            <Button
                aria-controls="customized-menu"
                aria-haspopup="true"
                variant="contained"
                color="primary"
                onClick={handleClick}
            >
                Activity
            </Button>
            <StyledMenu
                id="customized-menu"
                anchorEl={anchorEl}
                keepMounted
                open={Boolean(anchorEl)}
                onClose={handleClose}
            >
                <Link underline='none' href={'#/dashboard'}>
                    <StyledMenuItem>
                        <ListItemIcon>
                            <ListIcon />
                        </ListItemIcon>
                        <ListItemText primary="My routes" />
                    </StyledMenuItem>
                </Link>
                <Link underline='none' href={'#/create-route'}>
                    <StyledMenuItem>
                        <ListItemIcon>
                            <EditLocationIcon />
                        </ListItemIcon>
                        <ListItemText primary="Create route" />
                    </StyledMenuItem>
                </Link>
            </StyledMenu>
        </div>
    );
};

const StyledMenu = withStyles({
    paper: {
        border: '1px solid #d3d4d5',
    },
})((props) => (
    <Menu
        elevation={0}
        getContentAnchorEl={null}
        anchorOrigin={{
            vertical: 'bottom',
            horizontal: 'center',
        }}
        transformOrigin={{
            vertical: 'top',
            horizontal: 'center',
        }}
        {...props}
    />
));

const StyledMenuItem = withStyles((theme) => ({
    root: {
        '&:focus': {
            backgroundColor: theme.palette.secondary.main,
        },
    },
}))(MenuItem);


export default MisRutas;