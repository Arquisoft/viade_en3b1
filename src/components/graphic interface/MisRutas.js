import React from 'react';
import Button from '@material-ui/core/Button';
import { MenuItem, Paper, Popper, Grow, MenuList, ListItemIcon, Typography } from '@material-ui/core';
import ListIcon from '@material-ui/icons/List';
import EditLocationIcon from '@material-ui/icons/EditLocation';

const MisRutas = () => {
    const [open, setOpen] = React.useState(false);
    const anchorRef = React.useRef(null);

    const handleToggle = () => {
        setOpen((prevOpen) => !prevOpen);
    };

    return (
        <div>
            <Button
                ref={anchorRef}
                aria-controls={open ? 'menu-list-grow' : undefined}
                aria-haspopup="true"
                onClick={handleToggle}
                color="inherit"
            >
                My Routes
            </Button>
            <Popper open={open} anchorEl={anchorRef.current} transition disablePortal>
                {({ TransitionProps, placement }) => (
                    <Grow
                        {...TransitionProps}
                        style={{ transformOrigin: placement === 'bottom' ? 'center top' : 'center bottom' }}
                    >
                        <Paper >
                            <MenuList >

                                <MenuItem onClick={() => window.location.href='#/myroutes'}>
                                    <ListItemIcon>
                                        <ListIcon />
                                    </ListItemIcon>
                                    <Typography>See routes</Typography>
                                </MenuItem>

                                <MenuItem onClick={() =>  window.location.href='#/newRoute'}>
                                    <ListItemIcon>
                                        <EditLocationIcon />
                                    </ListItemIcon>
                                    <Typography>Create new</Typography>
                                </MenuItem>

                            </MenuList>
                        </Paper>
                    </Grow>
                )}
            </Popper>
        </div>
    );
};

export default MisRutas;