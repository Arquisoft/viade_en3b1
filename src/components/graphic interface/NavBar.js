import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';
import Login from "../services/login/Login.js";
import { LoggedOut, LoggedIn } from '@solid/react';
import { Link } from '@material-ui/core';
import "typeface-roboto";
import MisRutas from './MisRutas.js';
import logo1 from '../../assets/img/logo/logo1.svg';

const NavBar = () => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar >
          <Link underline='none' color='inherit' href={'#/welcome'} className={classes.title}>
            <img src={logo1} alt="Viade Logo" style={{ height: '3vh', width: '3vh' }} /> Viade
          </Link>

          <LoggedIn><MisRutas /></LoggedIn>
          <LoggedOut><Button className={classes.menuButton} href={"#/register"} color="inherit">Sign up</Button></LoggedOut>
          <Login />
        </Toolbar>
      </AppBar>
    </div>
  );
};

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
    fontSize: 20,
    letterSpacing: 1,
  },
  paper: {
    marginRight: theme.spacing(2),
  },
  menu: {
    width: 150,
  }
}));

export default NavBar;