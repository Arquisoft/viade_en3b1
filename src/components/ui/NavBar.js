import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';
import Login from "../btns/login/Login.js";
import { LoggedOut, LoggedIn } from '@solid/react';
import { Link, Typography, Grid } from '@material-ui/core';
import "typeface-roboto";
import MisRutas from '../btns/mis-rutas/MisRutas.js';
import logo1 from '../../assets/img/logo/logo_2.svg';

const NavBar = () => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar className={classes.toolbar}>

          {/* Home link */}
          <Grid className={classes.link}>
            <Grid container >
              <Link underline='none' color='inherit' href={'#/home'} >
                <Grid container>
                  <Grid item>
                    <img src={logo1} alt="Viade Logo" style={{ height: '4vh', width: '4vh' }} />
                  </Grid>
                  <Grid item>
                    <Typography variant='h4'>
                      Viade
                    </Typography>
                  </Grid>
                </Grid>
              </Link>
            </Grid>
          </Grid>

          <LoggedIn><MisRutas /></LoggedIn>

          <LoggedOut><Button className={classes.menuButton} href={"#/register"} color="inherit">Sign up</Button></LoggedOut>
          
          <Login />

        </Toolbar>
      </AppBar>
    </div>
  );
};

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  toolbar: {
    minHeight: '3vh',
    // alignItems: 'flex-start',
    paddingTop: theme.spacing(3),
    paddingBottom: theme.spacing(2),
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  link: {
    flexGrow: 1,
    // letterSpacing: 1,
  },
}));

export default NavBar;