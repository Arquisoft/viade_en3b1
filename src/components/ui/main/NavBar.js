import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';
import Login from "../../btns/login/Login.js";
import { LoggedOut, LoggedIn } from '@solid/react';
import { Link, Typography, Grid } from '@material-ui/core';
import "typeface-roboto";
import MisRutas from '../../btns/mis-rutas/MisRutas.js';
import ProfileMenu from '../../btns/profile-menu/ProfileMenu';
import logo1 from '../../../assets/img/logo/logo_2.svg';

const NavBar = () => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar className={classes.toolbar}>

          {/* Home link */}
          <Grid className={classes.link}>
            <Grid container >
              <Link underline='none' color='inherit' href={'#/home'} >
                <Grid container>
                  <Grid item>
                    <img src={logo1} alt="Viade Logo" style={{ height: '2.5rem', width: '2.5rem' }} />
                  </Grid>
                  <Grid item>
                    <Typography variant='h4'>
                      ViaDe
                    </Typography>
                  </Grid>
                </Grid>
              </Link>
            </Grid>
          </Grid>

          <LoggedIn><Button className={classes.menuButton} href={"#/social-feed"} color="primary" variant='contained'>Social Feed</Button></LoggedIn>
          <LoggedIn><MisRutas /></LoggedIn>

          <LoggedOut><Button className={classes.menuButton} href={"#/register"} color="primary" variant='contained'>Sign Up</Button></LoggedOut>
          <LoggedIn><ProfileMenu /></LoggedIn>
          
          <Login />

        </Toolbar>
      </AppBar>
    </div>
  );
};

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  toolbar: {
    minHeight: '3vh',
    // alignItems: 'flex-start',
    paddingTop: theme.spacing(3),
    paddingBottom: theme.spacing(2),
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  link: {
    flexGrow: 1,
    // letterSpacing: 1,
  },
}));

export default NavBar;