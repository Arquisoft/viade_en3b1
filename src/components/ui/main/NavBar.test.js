import NavBar from "./NavBar";
import { Typography, Button } from "@material-ui/core";
import MisRutas from "../../btns/mis-rutas/MisRutas";

test("Define NavBar", () => {
    expect(NavBar).toBeDefined();
});

test("Renders correctly", () => {
    const wrapper = render(<NavBar />);
    expect(wrapper).toMatchSnapshot();
});

test("Title ViaDe", () => {
    const wrapper = mount(<NavBar />);
    const title = wrapper.find(Typography).first().text();
    expect(title).toEqual("ViaDe");
});

test("Btn Social Feed", () => {
    const wrapper = mount(<NavBar />);
    const element = wrapper.find({ href: '#/social-feed' });

    // no session -> not exists
    expect(element.exists()).toBeFalsy();
});

test("Btn MisRutas", () => {
    const wrapper = mount(<NavBar />);
    const element = wrapper.find(MisRutas);
    expect(element).toBeDefined();

    // no session -> not exists
    expect(element.exists()).toBeFalsy();
});

test("Btn Login", () => {
    const wrapper = mount(<NavBar />);
    const login = wrapper.find({ href: '#/login' }).first().text();
    expect(login).toBeDefined();
    expect(login).toEqual("Sign In");
});

test("Btn Sign Up", () => {
    const wrapper = mount(<NavBar />);
    const element = wrapper.find({ href: '#/register' }).first().text();
    expect(element).toEqual("Sign Up");
});
