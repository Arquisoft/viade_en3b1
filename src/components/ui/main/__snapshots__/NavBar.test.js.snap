// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  class="makeStyles-root-1"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-2 MuiToolbar-gutters"
    >
      <div
        class="MuiGrid-root makeStyles-link-4"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
            href="#/home"
          >
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <img
                  alt="Viade Logo"
                  src="logo_2.svg"
                  style="height:2.5rem;width:2.5rem"
                />
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <h4
                  class="MuiTypography-root MuiTypography-h4"
                >
                  ViaDe
                </h4>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;
